src/init-ceph.in

```diff
@@ -21,6 +21,11 @@ fi
 SYSTEMD_RUN=$(which systemd-run 2>/dev/null)
 grep -qs systemd /proc/1/comm || SYSTEMD_RUN=""
 
+# Detect if this is inside docker, systemd-run fails inside containers.
+function detectDocker {
+readonly DOCKER_RUN=$(grep -q docker /proc/1/cgroup)
```

Comment:
:-1: This doesn't work on my system. I believe checking the existence of `/.dockerinit` will always work.